tcms/xmlrpc/tests/test_testplan.py

```diff
@@ -375,9 +377,74 @@ class TestImportCaseViaXML(test.TestCase):
     '''TODO: '''
 
 
-@unittest.skip('TODO: test case is not implemented yet.')
 class TestUpdate(test.TestCase):
-    '''TODO: '''
+    ''' Tests the XMLRPM testplan.update method '''
+    @classmethod
+    def setUpClass(cls):
+        super(TestUpdate, cls).setUpClass()
+
+        cls.user = UserFactory()
+        cls.http_req = make_http_request(user=cls.user)
+        perm_name = 'testplans.change_testplan'
+        user_should_have_perm(cls.http_req.user, perm_name)
+
+        cls.env_group_1 = TCMSEnvGroupFactory()
+        cls.env_group_2 = TCMSEnvGroupFactory()
+        cls.product = ProductFactory()
+        cls.version = VersionFactory(product=cls.product)
+        cls.tester = UserFactory()
+        cls.plan_type = TestPlanTypeFactory(name='manual smoking')
+        cls.plan_1 = TestPlanFactory(product_version=cls.version,
+                                     product=cls.product,
+                                     author=cls.tester,
+                                     type=cls.plan_type,
+                                     env_group=(cls.env_group_1,))
+        cls.plan_2 = TestPlanFactory(product_version=cls.version,
+                                     product=cls.product,
+                                     author=cls.tester,
+                                     type=cls.plan_type,
+                                     env_group=(cls.env_group_1,))
+
+    @classmethod
+    def tearDownClass(cls):
+        cls.plan_1.delete()
+        cls.plan_2.delete()
+        cls.plan_type.delete()
+        cls.tester.delete()
+        cls.user.delete()
+        cls.version.delete()
+        cls.product.delete()
+        cls.product.classification.delete()
+        cls.env_group_1.delete()
+        cls.env_group_2.delete()
+
+    def test_update_env_group(self):
+        # first verify that plan_1 and plan_2 point to self.env_group_1
+        self.assertEqual(self.plan_1.env_group.count(), 1)
+        self.assertEqual(self.plan_1.env_group.all()[0].pk, self.env_group_1.pk)
+
+        self.assertEqual(self.plan_2.env_group.count(), 1)
+        self.assertEqual(self.plan_2.env_group.all()[0].pk, self.env_group_1.pk)
+
+        # and there are only 2 objects in the many-to-many table
+        self.assertEqual(TCMSEnvPlanMap.objects.count(), 2)
```

Comment:
This is checking whether or not the initial condition is as we expect it to be but thinking about it it is probably not necessary. These lines will break only after modification of the setup methods and I guess the person doing that should be able to figure out that their change broke something and update the tests. I will remove these lines.